osgp/platform/osgp-core/src/test/java/org/opensmartgridplatform/core/application/services/DeviceNetworkAddressCleanupServiceTest.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright 2020 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.core.application.services;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.net.InetAddress;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opensmartgridplatform.domain.core.entities.Device;
+import org.opensmartgridplatform.domain.core.repositories.DeviceRepository;
+
+@ExtendWith(MockitoExtension.class)
+public class DeviceNetworkAddressCleanupServiceTest {
+
+    private DeviceRepository deviceRepository;
+    private DeviceNetworkAddressCleanupService deviceNetworkAddressCleanupService;
+
+    @BeforeEach
+    public void setUp() {
+        this.deviceRepository = Mockito.mock(DeviceRepository.class);
+        final boolean allowMultipleDevicesPerNetworkAddress = false;
+        final List<String> ipRangesAllowingMultipleDevicesPerAddress = Collections.emptyList();
+        this.deviceNetworkAddressCleanupService = new DeviceNetworkAddressCleanupService(this.deviceRepository,
```

Comment:
A bit strange to put this in setUp, while it is overridden in a lot of tests

Reply:
I changed the way the setup is handled, so this default before-each that is hardly ever used is no longer applied.